File: src/mobile/lib/app/utils/dto/rating/rating_item.dart
class RatingItem {
  RatingItem({
    required this.vendaId,
    required this.produtoId,
    required this.rating,
  });
  late final int vendaId;
  late final String produtoId;
  late final int rating;

  RatingItem.fromJson(Map<String, dynamic> json) {
    vendaId = json['vendaId'];
    produtoId = json['produtoId'];
    rating = json['rating'];
  }
}

File: src/services/Catalogo/Catalogo.API/Data/Entities/RatingItem.cs
using MongoDB.Bson.Serialization.Attributes;

namespace Catalogo.API.Data.Entities
{
  public class RatingItem
  {
    [BsonId]
    public string Id { get; set; } = Guid.NewGuid().ToString();

    public string ProdutoId { get; set; } = null!;

    public long VendaId { get; set; }

    public short Rating { get; set; }
  }
}
